thinc/layers/with_array.py

```diff
@@ -1,31 +1,88 @@
-from typing import Tuple, Callable, Optional, TypeVar, Union, cast
+from typing import Tuple, Callable, Optional, TypeVar, Union, cast, List
 
 from ..model import Model
 from ..config import registry
-from ..types import Array2d, Floats2d, Padded, Ragged, ArrayXd, Floats3d
-from ..types import List2d
-
-
-SeqT = TypeVar("SeqT", bound=Union[Padded, Ragged, List2d, ArrayXd])
+from ..types import (
+    Padded,
+    Ragged,
+    ArrayXd,
+    Array3d,
+    Floats1d,
+    Floats2d,
+    Floats3d,
+    Floats4d,
+    FloatsXd,
+    Ints1d,
+    Ints2d,
+    Ints3d,
+    Ints4d,
+    IntsXd,
+)
+
```

Comment:
Again, I couldn't find a solution involving any less code than this because you can't nest covariant types.